src/soft-delete-model.ts

```diff
@@ -1,8 +1,34 @@
-import {Document, SaveOptions} from "mongoose";
+import { Document, SaveOptions } from "mongoose";
 import * as mongoose from "mongoose";
 
+export interface SoftDelete {
+  isDeleted: boolean;
+  deletedAt: Date | null;
+}
+
+export type SoftDeleteDocument = SoftDelete & Document;
+
 export interface SoftDeleteModel<T extends Document> extends mongoose.Model<T> {
   findDeleted(): Promise<T[]>;
-  restore(query: Record<string, any>): Promise<{ restored: number }>;
-  softDelete(query: Record<string, any>, options?: SaveOptions): Promise<{ deleted: number }>;
```

Comment:
not sure if anyone is using the `options` parameter now.
this could be a breaking change

Reply:
Should we consider a major?

Reply:
yea, the majority is important.
In this case, if we bump the package version accordingly to introduce the changes, it should be fine.